job/job.go

```diff
@@ -89,3 +91,62 @@ func GetGenericJob(obj Object, config Config) (*batchv1.Job, error) {
 
 	return job, err
 }
+
+// SetConditionTrue tells the K8s controller at once that the status of the given Conditions is now "True"
+func (c *Config) SetConditionTrue(condition status.ConditionType) {
+	c.patchConditions(corev1.ConditionTrue, "", condition)
+}
+
+// SetConditionFalse tells the K8s controller at once that the status of the given Condition is now "True" and
+// provides the given message.
+// The arguments `message` and `args` follow the fmt.Sprintf() syntax.
+func (c *Config) SetConditionTrueWithMessage(condition status.ConditionType, message string, args ...interface{}) {
+	c.patchConditions(corev1.ConditionTrue, fmt.Sprintf(message, args...), condition)
+}
+
+// SetConditionFalse tells the K8s controller at once that the status of the given Condition is now "False" and
+// provides the given message.
+// The arguments `message` and `args` follow the fmt.Sprintf() syntax.
+func (c *Config) SetConditionFalse(condition status.ConditionType, message string, args ...interface{}) {
+	c.patchConditions(corev1.ConditionFalse, fmt.Sprintf(message, args...), condition)
+}
+
+// patchConditions patches the Status object on the K8s controller with the given Conditions
+func (c *Config) patchConditions(conditionStatus corev1.ConditionStatus, message string, conditions ...status.ConditionType) {
+	runtimeObject := c.Obj.GetRuntimeObject()
+	patch := client.MergeFrom(runtimeObject.DeepCopyObject())
+
+	for _, condition := range conditions {
+		c.Obj.GetStatus().Conditions.SetCondition(status.Condition{
+			Type:    condition,
+			Status:  conditionStatus,
+			Message: message,
+		})
+	}
+
+	err := c.Client.Status().Patch(c.CTX, c.Obj.GetRuntimeObject(), patch)
+	if err != nil {
+		c.Log.Error(err, "could not patch backup conditions")
+	}
+}
+
+// SetStarted sets the `c.Obj.GetStatus().Started` property to `true`.
+// In the same call to the k8s API it also sets all the given conditions to "True".
+// The arguments `message` and `args` follow the fmt.Sprintf() syntax.
+func (c *Config) SetStarted(trueCondition status.ConditionType, message string, args ...interface{}) {
+	runtimeObject := c.Obj.GetRuntimeObject()
+	patch := client.MergeFrom(runtimeObject.DeepCopyObject())
+
+	c.Obj.GetStatus().Started = true
+
+	c.Obj.GetStatus().Conditions.SetCondition(status.Condition{
+		Type:    trueCondition,
+		Status:  corev1.ConditionTrue,
+		Message: fmt.Sprintf(message, args...),
+	})
+
+	err := c.Client.Status().Patch(c.CTX, c.Obj.GetRuntimeObject(), patch)
```

Comment:
In another Operator we use `r.Client.Status().Update(..., ...)`. What is the difference to Patch()?

Reply:
As far as I understood, Patch will update just the properties that changed, whereas Update will replace the whole resource, i.e. update every property. That's a neat way to avoid the error about the resource not being up-to-date, because – at least for status updates – I don't really care if some other operator modified (parts of) the resource in the meantime. I just want to update that particular status field.